Utils/FaviconFetcher.swift

```diff
@@ -93,73 +94,84 @@ public class FaviconFetcher : NSObject, NSXMLParserDelegate {
         return deferred
     }
 
+    private func iconTypeForLink(rel:String) -> IconType {
+        var iconType = IconType.Guess
+        switch (rel) {
+        case "shortcut icon":
+            iconType = .Icon
+        case "icon":
+            iconType = .Icon
+        case "apple-touch-icon":
+            iconType = .AppleIcon
+        case "apple-touch-icon-precomposed":
+            iconType = .AppleIconPrecomposed
+        default:
+            iconType = IconType.Guess
+        }
+        return iconType
+    }
+
     // Loads and parses an html document and tries to find any known favicon-type tags for the page
     private func parseHTMLForFavicons(url: NSURL) -> Deferred<Maybe<[Favicon]>> {
+
         return fetchDataForURL(url).bind({ result -> Deferred<Maybe<[Favicon]>> in
             var icons = [Favicon]()
+            guard let data = result.successValue where result.isSuccess else {
+                return deferMaybe(icons)
+            }
 
-            if let data = result.successValue where result.isSuccess,
-               let element = RXMLElement(fromHTMLData: data) where element.isValid {
-                var reloadUrl: NSURL? = nil
-                element.iterate("head.meta") { meta in
-                    if let refresh = meta.attribute("http-equiv") where refresh == "Refresh",
-                        let content = meta.attribute("content"),
-                        let index = content.rangeOfString("URL="),
-                        let url = NSURL(string: content.substringFromIndex(index.startIndex.advancedBy(4))) {
-                            reloadUrl = url
-                    }
-                }
-
-                if let url = reloadUrl {
-                    return self.parseHTMLForFavicons(url)
-                }
+            let element = SWXMLHash.lazy(data)
+            var reloadUrl: NSURL? = nil
 
-                var bestType = IconType.NoneFound
-                element.iterateWithRootXPath("//head//link[contains(@rel, 'icon')]") { link in
-                    var iconType: IconType? = nil
-                    if let rel = link.attribute("rel") {
-                        switch (rel) {
-                            case "shortcut icon":
-                                iconType = .Icon
-                            case "icon":
-                                iconType = .Icon
-                            case "apple-touch-icon":
-                                iconType = .AppleIcon
-                            case "apple-touch-icon-precomposed":
-                                iconType = .AppleIconPrecomposed
-                            default:
-                                iconType = nil
-                        }
+            //lets see if this page requires a refresh to get the actual page
+            do {
+                let refreshNode = try element["html"]["head"]["meta"].withAttr("http-equiv", "Refresh").element
```

Comment:
I don't really have an example of a page with this behaviour. If you can link me to one I can confirm that this works.